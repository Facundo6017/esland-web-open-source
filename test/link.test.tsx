import { screen, render } from '@testing-library/preact'

import LinkReact from '@/components/Link';
import { expect } from 'vitest';

test('mount component with locale(ca) link', () => {
  render(<LinkReact href="/vota"
    locale="ca">This is a test</LinkReact>)
  const link = screen.getByRole('link');
  expect(link).toHaveAttribute('href', '/ca/vota')
  expect(link?.textContent).toBe('This is a test')
  expect(link?.outerHTML).toMatchSnapshot()
})

test('mount component with locale(default) link', () => {
  render(<LinkReact href="/vota">This is a test</LinkReact>)
  const link = screen.getByRole('link');
  expect(link).toHaveAttribute('href', '/vota')
  expect(link?.textContent).toBe('This is a test')
  expect(link?.outerHTML).toMatchSnapshot()
})

test('mount component with external link, locale not has effect', () => {
  expect(LinkReact).toBeTruthy();
  render(<LinkReact locale="a" href="https://www.google.com/">This is a test</LinkReact>)
  const link = screen.getByRole('link');
  expect(link).toHaveAttribute('href', 'https://www.google.com/')
  expect(link?.textContent).toBe('This is a test')
  expect(link?.outerHTML).toMatchSnapshot()

});

test('mount component with external link, locale not has effect', () => {
  expect(LinkReact).toBeTruthy();
  render(<LinkReact locale="ca" href="#more-d">This is a test</LinkReact>)
  const link = screen.getByRole('link');
  expect(link).toHaveAttribute('href', '#more-d')
  expect(link?.textContent).toBe('This is a test')
  expect(link?.outerHTML).toMatchSnapshot()

});
