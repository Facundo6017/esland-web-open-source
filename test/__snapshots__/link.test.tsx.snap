// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount component with external link, locale not has effect 1`] = `"<a href="https://www.google.com/">This is a test</a>"`;

exports[`mount component with external link, locale not has effect 2`] = `"<a href="#more-d">This is a test</a>"`;

exports[`mount component with locale(ca) link 1`] = `"<a href="/ca/vota">This is a test</a>"`;

exports[`mount component with locale(default) link 1`] = `"<a href="/vota">This is a test</a>"`;
